File: MarcenariaDotNet/Dtos/Geometrias/Geometria.cs
using System.Text.Json.Serialization;

namespace MarcenariaDotNet.Dtos.Geometrias;

[JsonPolymorphic(TypeDiscriminatorPropertyName = "geometria")]
[JsonDerivedType(typeof(Esfera), typeDiscriminator: "esfera")]
[JsonDerivedType(typeof(Cubo), typeDiscriminator: "cubo")]
[JsonDerivedType(typeof(Cilindro), typeDiscriminator: "cilindro")]
[JsonDerivedType(typeof(Paralelepipedo), typeDiscriminator: "paralelepipedo")]
//[JsonConverter(typeof(GeometriaConverter))]
public interface Geometria {
  public String GetGeometria();
  public String GetEstrutura();
  public double GetArea();
} 

File: codes/java/marcenaria-java-vertx/src/main/java/github/gmess/models/geometrias/Geometria.java
package github.gmess.models.geometrias;

import com.dslplatform.json.CompiledJson;

@CompiledJson(discriminator = "geometria")
public interface Geometria {
  String getEstrutura();

  String getGeometria();

  String getMaterial();

  double getArea();

  double getPreco();
}
